kv.go

```diff
@@ -155,11 +160,20 @@ type TxnStore interface {
 	NewTxn(readonly bool) Txn
 }
 
+// TxnReaderWriter contains the functions for reading and writing values within the store
+// and supports transactions.
+type TxnReaderWriter interface {
+	ReaderWriter
+
+	// NewTxn returns a new transaction.
+	NewTxn(readonly bool) Txn
```

Comment:
thought: It feels a little smelly that `NewTxn` is duplicated - it might be nice to remove this, possibly by declaring another interface that is embedded within `TxnStore` and `TxnReaderWriter` (unsure of name).  

Embedding `TxnReaderWriter` within `TxnStore` would also work but is less nice I think.

Reply:
I don't think there is anything wrong with having one or two methods that are present in different interfaces 🙂